dissect/target/plugins/apps/webservers/caddy.py

```diff
@@ -0,0 +1,138 @@
+import json
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import Iterator
+
+from dissect.util.ts import from_unix
+
+from dissect.target import plugin
+from dissect.target.exceptions import FileNotFoundError
+from dissect.target.helpers.fsutil import basename, open_decompress
+from dissect.target.plugins.apps.webservers.webservers import WebserverAccessLogRecord
+from dissect.target.target import Target
+
+LOG_FILE_REGEX = re.compile(r"(log|output file) (?P<log_file>.*)( \{)?$")
+LOG_REGEX = re.compile(
+    r'(?P<remote_ip>.*?) - - \[(?P<ts>\d{2}\/[A-Za-z]{3}\/\d{4}:\d{2}:\d{2}:\d{2} (\+|\-)\d{4})\] "(?P<method>.*?) (?P<uri>.*?) ?(?P<protocol>HTTP\/.*?)?" (?P<status_code>\d{3}) (?P<bytes_sent>-|\d+)'  # noqa: E501
+)
+
+
+class CaddyPlugin(plugin.Plugin):
+    __namespace__ = "caddy"
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.log_paths = self.get_log_paths()
+
+    @plugin.internal
+    def get_log_paths(self) -> list[Path]:
+        log_paths = []
+
+        # Add any well known default Caddy log locations
+        log_paths.extend(self.target.fs.path("/var/log").glob("caddy_access.log*"))
+
+        # Check for custom paths in Caddy config
+        if (config_file := self.target.fs.path("/etc/caddy/Caddyfile")).exists():
+            found_roots = []
+            for line in config_file.open("rt"):
+                line = line.strip()
+                if not line or line.startswith("#"):
```

Comment:
What about Caddy, can you also fix it here? Seems a bit more complicated unfortunately.

Reply:
Added in [bd39b7a](https://github.com/fox-it/dissect.target/pull/127/commits/bd39b7a4ec215638055a802d5044797d027c3bd3).

Reply:
Slightly changed it :)